### DISCUSSION TITLE
Minimum Magnification Requirements

### DISCUSSION THREAD

**Discussion by murrageo** [neuropoly/axondeepseg#566] [Help needed] [answered]
Hello all, 

I'm curious whether or not ADS can be made to work with light microscopy images taken at 20X?  Our image resolution is sufficient to distinguish axons, myelin, and background visually with ease, and creating a mask manually would certainly be possible.  Our goal is to find metrics like axon number, area, myelin thickness, and g-ratio from stitched images of whole femoral nerve branches.  It seems to me that training an ADS model for this use case would be possible, but I am not the most technically versed.  I do not want to spend the time creating the masks for training manually if there is a technical hangup, i.e. pixel width ranges, that ADS cannot overcome even by new model training.  Does this goal seem possible by training a new model? 

Thanks very much, 
M

**Comment by mathieuboudreau** [comment, marked as answer]
Hi @murrageo,

Any chance you could share a sample image?

**Reply by murrageo** [reply]
Hi @mathieuboudreau ,

Sure.  I'll attach three.  We're hoping that our model will be able to accurately quantify each of the metrics I mentioned previously from nerve histology of mouse models affected by a variety of inherited neuropathies.  

Best 

![2](https://user-images.githubusercontent.com/50328820/133278579-e2de28a0-e98a-4d1d-966f-70cc41304e20.png)
![1](https://user-images.githubusercontent.com/50328820/133278594-9d360b45-6e5c-4ea9-8a63-9150bbcce174.png)
![3](https://user-images.githubusercontent.com/50328820/133278609-373490e1-a837-43bd-9024-20d7ba6e375e.png)

**Reply by hermancollin** [reply]
Hi @murrageo

I played around with the last image you sent us. This is the kind of mask you would obtain using ADS. I used our default TEM model with a pixelsize of 0.015.
(I would suggest to overlay one image on top of the other to assess the segmentation quality)
<details>
  <summary>image + mask</summary>

![bigmurr_seg-axonmyelin](https://user-images.githubusercontent.com/83031821/133288360-d6f585b0-ec81-42df-92af-3a463ff974c8.png) 
![bigmurr](https://user-images.githubusercontent.com/83031821/133288342-1659f53d-c90f-4745-abb3-bf31f6adf6a8.png)
</details>

Then, when morphometrics are computed, the software generates an index image (see below) which enables you to filter out the little glitches you can see here and there from your final analysis.

<details>
  <summary>Index file</summary>

![Screenshot_20210914_113948](https://user-images.githubusercontent.com/83031821/133289189-3ca04459-5552-49ef-a034-f3d6a4d31ce3.png)
</details>

Please keep in mind that this is a very preliminary result. I suspect the segmentation quality could easily be improved by comparing our default models and playing around with the pixel_size parameter. Of course, training a new model would probably give you the most accurate results, but as you know this requires some time.
Concerning the output metrics, axon number, area, myelin thickness and g-ratio are all supported by ADS.

**Reply by murrageo** [reply]
Thanks for your reply!  

I've tried changing the pixel sizes around a bit and a few different methods of pre-processing.  I'm fairly happy with the percent error I'm seeing for Wildtype mice: 1-5%  (based on manual nerve counts that we're treating as ground truth).  Error seems greater for mice with neuropathy:  10-20%.  On the higher end, that's more than we can tolerate.  I hadn't tried filtering glitches on the index image, and so I'll give that a try.  I'm in the process of evaluating ADS output against manually quantified myelin thickness and g-ratio.  I'll report back when I know how that turns out.

Thanks again
